langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+            String embeddingId = (String) searchDocument.get(DEFAULT_FIELD_ID);
+            List<Double> embeddingList = (List<Double>) searchDocument.get(DEFAULT_FIELD_CONTENT_VECTOR);
+            float[] embeddingArray = doublesListToFloatArray(embeddingList);
+            Embedding embedding = Embedding.from(embeddingArray);
+            String embbededContent = (String) searchDocument.get(DEFAULT_FIELD_CONTENT);
+            EmbeddingMatch<TextSegment> embeddingMatch;
+            if (isNotNullOrBlank(embbededContent)) {
+                LinkedHashMap metadata = (LinkedHashMap) searchDocument.get(DEFAULT_FIELD_METADATA);
+                List attributes = (List) metadata.get(DEFAULT_FIELD_METADATA_ATTRS);
+                Map<String, String> attributesMap = new HashMap<>();
+                for (Object attribute : attributes) {
+                    LinkedHashMap innerAttribute = (LinkedHashMap) attribute;
+                    String key = (String) innerAttribute.get("key");
+                    String value = (String) innerAttribute.get("value");
+                    attributesMap.put(key, value);
+                }
+                Metadata langChainMetada = Metadata.from(attributesMap);
+                TextSegment embedded = TextSegment.textSegment(embbededContent, langChainMetada);
+                embeddingMatch = new EmbeddingMatch<>(score, embeddingId, embedding, embedded);
+            } else {
+                embeddingMatch = new EmbeddingMatch<>(score, embeddingId, embedding, null);
+            }
+            result.add(embeddingMatch);
+        }
+        return result;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(
+                singletonList(id),
+                singletonList(embedding),
+                embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(
+            List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("Empty embeddings - no ops");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(),
+                "embeddings size is not equal to embedded size");
+
+        List<Document> searchDocuments = new ArrayList<>();
+        for (int i = 0; i < ids.size(); ++i) {
+            Document document = new Document();
+            document.setId(ids.get(i));
+            document.setContent_vector(floatsArrayToList(embeddings.get(i).vector()));
+            if (embedded != null) {
+                document.setContent(embedded.get(i).text());
+                Document.Metadata metadata = new Document.Metadata();
+                metadata.setSource("langchain4j");
```

Comment:
could you please explain what is the purpose of this field?

Reply:
The "metadata source" is configured in the index in other languages (like JavaScript, in the commit linked at the top of this issue), but it doesn't seemed to be used. As I guessed it was the source of the data, I put "langchain4j" but indeed that's useless.
I've removed that line, but kept the field in the index, to have the same index configuration as in the other languages.